django_project/changes/views/sponsor.py

```diff
@@ -659,3 +663,57 @@ def generate_sponsor_cloud(request, **kwargs):
         context={
             'image': image_path,
             'the_project': project})
+
+
+class GenerateSponsorPDFView(SponsorMixin, LoginRequiredMixin, TemplateView):
+    """Template View for invoice generation"""
+    context_object_name = 'sponsors'
+    template_name = 'sponsor/invoice.html'
+
+    def get_context_data(self, **kwargs):
+        """Get the context data which is passed to a template.
+
+        :param kwargs: Any arguments to pass to the superclass.
+        :type kwargs: dict
+
+        :returns: Context data which will be passed to the template.
+        :rtype: dict
+        """
+        context = super(GenerateSponsorPDFView, self).\
+            get_context_data(pagesize="A4", **kwargs)
+        project_slug = self.kwargs.get('project_slug', None)
+        sponsor_slug = self.kwargs.get('slug', None)
+        sponsors = SponsorshipPeriod.approved_objects.all()
+
+        context['project_slug'] = project_slug
+        context['sponsor_slug'] = sponsor_slug
+        context['sponsors'] = sponsors
+        if project_slug and sponsor_slug:
+            project = Project.objects.get(slug=project_slug)
+            context['sponsor'] = sponsors.get(
+                project=project,
+                slug=sponsor_slug)
+            context['project'] = project
+            context['invoice_no'] = '12345'
```

Comment:
this needs to be updated. Perhaps give each model object an ID and use that?

Reply:
Thanks ill update that shortly.

Reply:
I think we might also need to be able to manually enter an invoice number...

Reply:
Alright @timlinux ill do just that. 
Thanks